### PR TITLE
ENH: Relocate the `sim_response` method to allow re-use

### PR DESCRIPTION
**Pull request by jhlegarreta**
Relocate the `sim_response` method to allow re-use.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Following [the note](https://github.com/nipy/dipy/issues/718#issuecomment-513961186) in issue #718.

@ShreyasFadnavis @skoudoro @arokem @Garyfallidis It is probably easier to comment once the PR is in place.

Please suggest changes (make documentation more accurate, move to `sim` as suggested by Shreyas, etc.) as necessary.

**Comment by ShreyasFadnavis** [comment]
@jhlegarreta Are you working on moving it to `sims` ? If you are busy, I can create a PR here! I can then use this in the example.. Let me know 👍

**Comment by jhlegarreta** [comment]
@ShreyasFadnavis I was waiting for a consensus answer concerning potentially moving the method to the `sims` (or other) package.

I put it in `mcsd` following the rationale of `auto_response`, which is in `reconst.csdeconv`, but I would be fine with whatever it is decided, and can easily put it to `sims` if that is preferable.

**Comment by ShreyasFadnavis** [comment]
@jhlegarreta .. Lol.. I think I edited your comment by mistake.. can you please rewrite it? I'm terribly sorry!

**Comment by arokem** [comment]
@shreyas : It’s in your email inbox (but not in Jon’s)

**Comment by ShreyasFadnavis** [comment]
> @ShreyasFadnavis I was waiting for a consensus answer concerning potentially moving the method to the sims (or other) package.
> 
> I put it in mcsd following the rationale of auto_response, which is in reconst.csdeconv, but I would be fine with whatever it is decided, and can easily put it to sims if that is preferable.

Gotcha! I guess lets stick to the current design then.. this PR LGTM! @skoudoro @arokem @Garyfallidis any suggestions? This can go into 1.0 maybe? :)

**Comment by jhlegarreta** [comment]
Ariel is right. But one can see the history and editions of the comments clicking on the `Edited by {...}` field in the corresponding GitHub comment. Thanks both !

**Comment by jhlegarreta** [comment]
**CI errors seem to be unrelated**:
` test_particle_filtering_traking_workflows` and `test_local_fiber_tracking_workflow` both fail with
```
ValueError: cannot expose native-only dtype 'g' in non-native byte order '<'
via buffer interface
```
according to [this log line](https://ci.appveyor.com/project/skoudoro/dipy-5dd9b/builds/26295278/job/4ol3l54arwrh291x#L5895) and [this one](https://ci.appveyor.com/project/skoudoro/dipy-5dd9b/builds/26295278/job/4ol3l54arwrh291x#L5974).

**Comment by arokem** [comment]
The name suggests that this should go into the sims module. 

More-so -- are you sure that there is nothing in the sims module that already does this?

**Comment by jhlegarreta** [comment]
> The name suggests that this should go into the sims module.

@arokem done in e76a4e9 . Since none of the methods in that module has the `sims_*` prefix, I also renamed it. Please check if the new name is more meaningful.

> More-so -- are you sure that there is nothing in the sims module that already does this?

I have gone through the methods, and would dare to say that none does this.

**Comment by jhlegarreta** [comment]
CI is reporting
```
E   ImportError: cannot import name 'default_sphere'
```

which I find weir, since the method is present in `dipy.data` and this PR has not modified it.

**Comment by ShreyasFadnavis** [comment]
> CI is reporting
> 
> ```
> E   ImportError: cannot import name 'default_sphere'
> ```
> 
> which I find weird, since the method is present in `dipy.data` and this PR has not modified it.

+1 ... I second @jhlegarreta that no other method in sims does this..

**Comment by ShreyasFadnavis** [comment]
> > The name suggests that this should go into the sims module.
> 
> @arokem done in [e76a4e9](https://github.com/nipy/dipy/commit/e76a4e9c4543f65811d67603d288a6bece5ab0a8) . Since none of the methods in that module has the `sims_*` prefix, I also renamed it. Please check if the new name is more meaningful.
> 
> > More-so -- are you sure that there is nothing in the sims module that already does this?
> 
> I have gone through the methods, and would dare to say that none does this.

+1 I second @jhlegarreta that no other function does this..

**Comment by arokem** [comment]
@jhlegarreta : I think that the issue is that we now have circular dependencies, because dipy.sims.voxel imports dipy.data, which imports dipy.sims.voxel ([here](https://github.com/nipy/dipy/blob/master/dipy/data/__init__.py#L17)).

One way to deal with this is not to import the default sphere from dipy.data at the module level and instead require a sphere as input, with a local import of the default sphere if None is provided. 

I can add suggestions to the code for this.

File: dipy/sims/voxel.py [hunk 1]
```diff
@@ -995,3 +999,52 @@ def multi_tensor_msd(mf, mevals=None, tau=1 / (4 * np.pi ** 2)):
...
+    bvals = np.array(bvals, copy=True)
+    evecs = np.zeros((3, 3))
+    z = np.array([0, 0, 1.])
+    evecs[:, 0] = z
+    evecs[:2, 1:] = np.eye(2)
+
+    n = np.arange(0, sh_order + 1, 2)
+    m = np.zeros_like(n)
+
+    big_sphere = default_sphere.subdivide()
```
Comment (arokem) [inline dipy/sims/voxel.py]:
```suggestion
    if sphere is None:
        from dipy.data import default_sphere
    big_sphere = default_sphere.subdivide()
```

File: dipy/sims/voxel.py [hunk 2]
```diff
@@ -995,3 +999,52 @@ def multi_tensor_msd(mf, mevals=None, tau=1 / (4 * np.pi ** 2)):
     for j, f in enumerate(mf):
         msd += f * single_tensor_msd(mevals[j], tau=tau)
     return msd
+
+
+def multi_shell_fiber_response(sh_order, bvals, evals, csf_md, gm_md):
```
Comment (arokem) [inline dipy/sims/voxel.py]:
```suggestion
def multi_shell_fiber_response(sh_order, bvals, evals, csf_md, gm_md, sphere=None):
```

File: dipy/sims/voxel.py (hunk 1 above)
Comment (arokem) [reply dipy/sims/voxel.py]: Edited to be more sensible.

**Comment by arokem** [comment]
Alternatively, we could edit the dipy.data model to import dipy.sims locally within the one function where that function is used, so [this](https://github.com/nipy/dipy/blob/master/dipy/data/__init__.py#L17), would be moved [here](https://github.com/nipy/dipy/blob/master/dipy/data/__init__.py#L328)

**Comment by jhlegarreta** [comment]
Thanks @arokem !! I went on with your first suggestion in  358ec41.

File: dipy/sims/voxel.py (hunk 1 above)
Comment (jhlegarreta) [reply dipy/sims/voxel.py]: Done in 358ec41.

File: dipy/sims/voxel.py (hunk 2 above)
Comment (jhlegarreta) [reply dipy/sims/voxel.py]: Done in 358ec41.

**Comment by jhlegarreta** [comment]
@arokem is there still something missing so that it can be merged?

**Comment by arokem** [comment]
No - sorry - this looks good to me. 

Does anyone else want to take a look, before we merge?

**Comment by skoudoro** [comment]
Travis and all CI's are failing, I do not think we can merge.

Need to look at what is happening

**Comment by jhlegarreta** [comment]
Thanks both. Let's see if ad8aab8 makes the trick !

**Comment by jhlegarreta** [comment]
It made the trick. Thanks @arokem ! :fireworks: :tada: :rocket:

**Comment by arokem** [comment]
Yeah - this looks fine to me. @skoudoro : does this look OK to you now?

**Comment by skoudoro** [comment]
Travis Happy = I'm happy!  😄

+1 for merging

**Pull request merged**